grype/matcher/rpm/almalinux.go

```diff
@@ -0,0 +1,299 @@
+// 1. Use RHEL vulnerability disclosures as the base (similar to RHEL EUS)
+// 2. Filter results using AlmaLinux-specific unaffected package records
+// 3. Handle source RPM to binary RPM relationships when matching unaffected records
+//
+// The matching process:
+// 1. Find RHEL disclosures that match the package (treating AlmaLinux as RHEL-compatible)
+// 2. Find AlmaLinux unaffected packages that apply to this package (including source/binary relationships)
+// 3. Remove vulnerabilities that are marked as unaffected in AlmaLinux
+func almaLinuxMatches(provider result.Provider, searchPkg pkg.Package) ([]match.Match, error) {
+	if strings.HasSuffix(searchPkg.Name, "-debuginfo") || strings.HasSuffix(searchPkg.Name, "-debugsource") {
+		return nil, nil // almaloinux explicitly never publishes advisories for RPMs that are only debug material
+		// consider these as fixed; otherwise we will have no fixed version for them, and they will be considered
+		// to be affected by every CVE that affects their src rpm at any version.
+	}
+	// Create a RHEL-compatible distro for finding base disclosures
+	rhelCompatibleDistro := *searchPkg.Distro
+	rhelCompatibleDistro.Type = distro.RedHat // treat as RHEL for disclosure lookup
+
+	pkgVersion := version.New(searchPkg.Version, pkg.VersionFormat(searchPkg))
+
+	// Step 1: Find RHEL disclosures for the package
+	disclosures, err := provider.FindResults(
+		search.ByPackageName(searchPkg.Name),
+		search.ByDistro(rhelCompatibleDistro), // look for RHEL disclosures
+		internal.OnlyQualifiedPackages(searchPkg),
+		internal.OnlyVulnerableVersions(pkgVersion),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("matcher failed to fetch RHEL disclosures for AlmaLinux pkg=%q: %w", searchPkg.Name, err)
+	}
+
+	if len(disclosures) == 0 {
+		return nil, nil
+	}
+
+	// Step 2: Find AlmaLinux unaffected packages that apply to this package
+	unaffectedResults, err := provider.FindResults(
+		search.ByPackageName(searchPkg.Name),
+		search.ByExactDistro(*searchPkg.Distro), // use exact AlmaLinux distro for unaffected lookup (no aliases)
+		internal.OnlyQualifiedPackages(searchPkg),
+		search.ForUnaffected(),
+	)
+	if err != nil {
+		log.WithFields("error", err, "distro", searchPkg.Distro, "pkg", searchPkg.Name).Debug("failed to fetch AlmaLinux unaffected packages")
+		// If we can't get unaffected data, return the original disclosures
+		return disclosures.ToMatches(), nil
+	}
+
+	// Step 3: Also look for unaffected packages using source/binary RPM relationships
+	relatedUnaffectedResults := findRelatedUnaffectedPackages(provider, searchPkg)
+	// Merge the related results into the main unaffected results
+	for key, results := range relatedUnaffectedResults {
+		unaffectedResults[key] = append(unaffectedResults[key], results...)
+	}
+
+	// Step 4: Update RHEL disclosures with AlmaLinux-specific fix information
+	updatedDisclosures := updateDisclosuresWithAlmaLinuxFixes(disclosures, unaffectedResults, pkgVersion)
```

Comment:
I suspect (but have not proven this out) that steps 3 and 4 could leverage the result [merge](https://github.com/anchore/grype/blob/main/grype/matcher/internal/result/results.go#L130-L133) and [filter](https://github.com/anchore/grype/blob/main/grype/matcher/internal/result/results.go#L216) functions.